src/lib/pyth.ts

```diff
@@ -0,0 +1,744 @@
+								type: 'uint256'
+							}
+						],
+						internalType: 'struct PythStructs.Price',
+						name: 'price',
+						type: 'tuple'
+					},
+					{
+						components: [
+							{
+								internalType: 'int64',
+								name: 'price',
+								type: 'int64'
+							},
+							{
+								internalType: 'uint64',
+								name: 'conf',
+								type: 'uint64'
+							},
+							{
+								internalType: 'int32',
+								name: 'expo',
+								type: 'int32'
+							},
+							{
+								internalType: 'uint256',
+								name: 'publishTime',
+								type: 'uint256'
+							}
+						],
+						internalType: 'struct PythStructs.Price',
+						name: 'emaPrice',
+						type: 'tuple'
+					}
+				],
+				internalType: 'struct PythStructs.PriceFeed[]',
+				name: 'priceFeeds',
+				type: 'tuple[]'
+			}
+		],
+		stateMutability: 'payable',
+		type: 'function'
+	},
+	{
+		inputs: [
+			{
+				internalType: 'bytes[]',
+				name: 'updateData',
+				type: 'bytes[]'
+			}
+		],
+		name: 'updatePriceFeeds',
+		outputs: [],
+		stateMutability: 'payable',
+		type: 'function'
+	},
+	{
+		inputs: [
+			{
+				internalType: 'bytes[]',
+				name: 'updateData',
+				type: 'bytes[]'
+			},
+			{
+				internalType: 'bytes32[]',
+				name: 'priceIds',
+				type: 'bytes32[]'
+			},
+			{
+				internalType: 'uint64[]',
+				name: 'publishTimes',
+				type: 'uint64[]'
+			}
+		],
+		name: 'updatePriceFeedsIfNecessary',
+		outputs: [],
+		stateMutability: 'payable',
+		type: 'function'
+	}
+] as const;
+
+export const DEFAULT_PYTH_ADDR = '0xff1a0f4744e8582DF1aE09D5611b887B6a12925C' as const;
+
+const HERMES_V2 = 'https://hermes.pyth.network/v2/updates/price/latest';
+const U64_MAX = (1n << 64n) - 1n;
+
+type PublishTimeLike = bigint | number | string | null | undefined;
+
+type ParsedPriceRow = {
+	price_id?: string;
+	priceId?: string;
+	id?: string;
+	price_feed?: { id?: string };
+	publish_time?: PublishTimeLike;
+	publishTime?: PublishTimeLike;
+	price?: {
+		publish_time?: PublishTimeLike;
+		publishTime?: PublishTimeLike;
+	};
+};
+
+type ParsedSection = {
+	price_updates?: ParsedPriceRow[];
+	updates?: ParsedPriceRow[];
+} | null;
+
+type HermesJson = {
+	binary?: unknown;
+	priceUpdates?: unknown;
+	updates?: unknown;
+	parsed?: ParsedSection;
+};
+
+const coerceHexArray = (value: unknown): `0x${string}`[] => {
+	if (!Array.isArray(value)) return [];
+	const result: `0x${string}`[] = [];
+	for (const item of value) {
+		if (typeof item === 'string' && item.length) {
+			result.push((item.startsWith('0x') ? item : `0x${item}`) as `0x${string}`);
+		} else if (item && typeof item === 'object' && 'data' in (item as Record<string, unknown>)) {
+			const candidate = (item as { data?: string }).data;
+			if (typeof candidate === 'string' && candidate.length) {
+				result.push((candidate.startsWith('0x') ? candidate : `0x${candidate}`) as `0x${string}`);
+			}
+		}
+	}
+	return result;
+};
+
+const normalizeIdsNo0x = (ids: `0x${string}`[]): string[] =>
```

Comment:
this looks like redundant as similar fn should be available in libs like viem if im not mistaken, actually many of the functions in this file seem to be like that